taxi/forms.py

```diff
@@ -0,0 +1,75 @@
+from django import forms
+from django.contrib.auth.forms import UserCreationForm, UserChangeForm
+from django.core.exceptions import ValidationError
+from django.core.validators import MaxLengthValidator, RegexValidator
+
+from taxi.models import Driver, Car
+
+
+class CarCreateForm(forms.ModelForm):
+    drivers = forms.ModelMultipleChoiceField(
+        queryset=Driver.objects.all(),
+        widget=forms.CheckboxSelectMultiple(),
+        required=False,
+    )
+
+    class Meta:
+        model = Car
+        fields = "__all__"
+
+
+class DriverCreateForm(UserCreationForm):
+    cars = forms.ModelMultipleChoiceField(
+        queryset=Car.objects.all(),
+        widget=forms.CheckboxSelectMultiple(),
+        required=False,
+    )
+    license_number = forms.CharField(
+        validators=[
+            MaxLengthValidator(8),
+            RegexValidator(
+                regex=r"^[A-Z]{3}",
+                message="The first 3 characters must be uppercase letters",
+                code="invalid_first_three_characters"
+            ),
+            RegexValidator(
+                regex=r"\d{5}$",
+                message="The last 5 characters must be numbers",
+                code="invalid_last_five_characters"
+            )
+        ]
+    )
+
+    class Meta(UserCreationForm.Meta):
+        model = Driver
+        fields = (
+            "username", "first_name", "last_name", "email", "license_number"
+        )
+
+    def clean_license_number(self) -> str:
+        license_number = self.cleaned_data["license_number"]
+        if Driver.objects.filter(license_number=license_number).exists():
+            raise ValidationError(
+                "A driver with this license number already exists."
+            )
+        return license_number
+
+
+class DriverLicenseUpdateForm(DriverCreateForm):
+    def __init__(self, *args, **kwargs):
```

Comment:
no need to override init - overriding fields in Meta class would be enough

Reply:
Without init and save, it doesn't work correctly. I have changed the logic by separating the validation into its own function.